templates/puppet_certs_common.ps1

```diff
@@ -13,6 +13,18 @@ Function Out-Certificate {
 }
 
 Function Import-CSharpCode {
+
+  # When puppet is run via 'puppet agent -t', or via the service, when the
+  # account used to execute the run is the 'Local System' account, the temp
+  # directory is set to the value seen below in the if test. When this happens
+  # the Add-Type commandlet cannot compile the C# code in the type definition.
+  # The compilation process requires creating temp files, and that process is
+  # broken if the temp directory is not fixed first.
+  if( $env:tmp -eq 'C:\Windows\system32\config\systemprofile\AppData\Local\Temp') {
```

Comment:
Maybe use `$env:WINDIR` instead of hardcoding c:\?

Reply:
Good idea.